scripts/check_setting_visibility.py

```diff
@@ -0,0 +1,180 @@
+        # The basic settings should be in advanced, and advanced in expert
+        for visibility_type in setting_visibility:
+
+            # Skip the basic because it cannot be compared to previous list
+            if visibility_type == 'basic':
+                continue
+
+            all_settings_in_this_type = []
+            not_valid_setting_by_order[visibility_type] = []
+
+            item = setting_visibility_items[visibility_type]
+            for category, category_settings in item.items():
+                all_settings_in_this_type.extend(category_settings)
+
+
+            for setting in visible_settings_order:
+                if setting not in all_settings_in_this_type:
+                    not_valid_setting_by_order[visibility_type].append(setting)
+
+
+        # If any of the settings is defined not correctly then the test is failed
+        has_invalid_settings = False
+
+        for type, settings in not_valid_categories.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following categories are defined incorrectly")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect categories : '%s'" % (settings))
+                print()
+
+
+
+        for type, settings in not_valid_setting_by_category.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following settings do not exist anymore in fdmprinter definition or in wrong category")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect settings : '%s'" % (settings))
+                print()
+
+
+        for type, settings in not_valid_setting_by_order.items():
+            if len(settings) > 0:
+                has_invalid_settings = True
+                print("The following settings are defined in the incorrect order in setting visibility definitions")
+                print("  Visibility type : '%s'" % (type))
+                print("  Incorrect settings : '%s'" % (settings))
+
+        return has_invalid_settings
+
+
+if __name__ == "__main__":
+
+    all_setting_visibility_files = glob.glob(os.path.join(os.path.join(SCRIPT_DIR, "..", "resources", "setting_visibility"), '*.cfg'))
+    fdmprinter_def_path = os.path.join(SCRIPT_DIR, "..", "resources", "definitions", "fdmprinter.def.json")
+
+    inspector = SettingVisibilityInspection()
+    inspector.defineAllCuraSettings(fdmprinter_def_path)
+
+    setting_visibility_items = {}
+    for file_path in all_setting_visibility_files:
+        temp = inspector.getSettingsFromSettingVisibilityFile(file_path)
+
+        base_name = os.path.basename(file_path)
+        visibility_type = base_name.split('.')[0]
+
+        setting_visibility_items[visibility_type] = temp
+
+    has_invalid_settings = inspector.validateSettingsVisibility(setting_visibility_items)
+
+    sys.exit(0 if not has_invalid_settings else 1)
```

Comment:
Just use PyTest to fail the test then.

Reply:
There was a reason why I followed this approach, in CI it will catch the output and show "fail" or "pass"